src/components/ContractTab/FunctionInterface.vue

```diff
@@ -187,6 +180,62 @@ export default {
             'isNative',
             'nativeAccount',
         ]),
+        inputComponents() {
+            if (!Array.isArray(this.abi?.inputs))
+                return [];
+
+            const getExtraBindingsForType = ({ type, name }, index) => {
+                const label = `${name ? name : `Param ${index + 1}`}`;
+                const extras = {};
+
+                // represents integer bits (e.g. uint256) for int types, or array length for array types
+                let size = undefined;
+                if (parameterIsArrayType(type)) {
+                    size = getExpectedArrayLengthFromParameterType(type)
+                } else if (parameterIsIntegerType(type)) {
+                    size = getIntegerBits(type)
+                }
+
+                const getIntSize = () => type.match(/\d+(?=\[)/)[0];
+
+                if (parameterTypeIsUnsignedIntArray(type)) {
+                    extras['uint-size'] = getIntSize();
+                } else if (parameterTypeIsSignedIntArray(type)) {
+                    extras['int-size'] = getIntSize();
+                }
+
+                const defaultModelValue = parameterTypeIsBoolean(type) ? null : '';
+
+                return {
+                    ...extras,
+                    label,
+                    size,
+                    modelValue: this.inputModels[index] ?? defaultModelValue,
```

Comment:
I didn't know this `??` operator. I always used `||` instead, but now (after some research) I understand is not the same. Good to know, I will incorporate it in my writing.